src/loader.ts

```diff
@@ -0,0 +1,27 @@
+import { readFile } from "node:fs/promises";
+import type { LoadFnOutput, LoadHookContext } from "node:module";
+import { fileURLToPath } from "node:url";
+import { transformSync } from "./index.ts";
+
+type NextLoad = (
+	url: string,
+	context?: LoadHookContext,
+) => LoadFnOutput | Promise<LoadFnOutput>;
+
+export async function load(
+	url: string,
+	context: LoadHookContext,
+	nextLoad: NextLoad,
+) {
+	const { format } = context;
+	if (format.includes("typescript")) {
+		const source = await readFile(fileURLToPath(url), "utf8");
+		const { code } = transformSync(source.toString(), { mode: "strip-only" });
+		return {
+			format: format.replace("-typescript", ""),
+			shortCircuit: true,
```

Comment:
You shouldn't use `readFile` in case there are more loaders chained with this one.
```suggestion
		const finalFormat = format.replace("-typescript", "");
		const source = await nextLoad(url, { ...context, format: finalFormat });
		const { code } = transformSync(source.toString(), { mode: "strip-only" });
		return {
			format: finalFormat,
```

Reply:
apparently it must use the `esm` loader as commonjs will return `undefined` source

Reply:
Ah right, I forgot about that. Yeah, I think it's fine to request `nextLoad` with format `module`